pennylane/tape/tape.py

```diff
@@ -237,6 +237,47 @@ def stop_at(obj):  # pylint: disable=unused-argument
     return new_tape
 
 
+def expand_tape_state_prep(tape, skip_first=True):
+    """Expand all instances of StatePrep operations in the tape.
+
+    Args:
+        tape (QuantumTape): The tape to expand.
+        skip_first (Bool): If ``True``, will not expand a StatePrep operation if
```

Comment:
Is this needed or just provided for extra flexibility?

Reply:
Just for flexibility if we need to expand everything in the future,

Reply:
I think my point is why `skip_first=False` is needed at all.

Reply:
Right now `skip_first=True` will expand all StatePrep operations except for the first one. Suppose someone creates a new simulator device where they have no support for StatePrep operations, then they need to expand all of them, so they would just copy the code for our expand function but set `skip_first=False`. 

I don't see it being used very much, I don't mind removing it if everyone thinks it's extra.

Reply:
No, fine to keep it.